packages/cloud_firestore/lib/src/collection_reference.dart

```diff
@@ -39,6 +39,17 @@ class CollectionReference extends Query {
     }
     return new DocumentReference._(_firestore, childPath);
   }
+
+  /// Returns a `DocumentReference` with an auto-generated ID, after
+  /// populating it with provided [data].
+  ///
+  /// The unique key generated by is prefixed with a client-generated timestamp
+  /// so that the resulting list will be chronologically-sorted.
+  Future<DocumentReference> add(Map<String, dynamic> data) async {
```

Comment:
I think the add method should also accept a `[String path]` to pass it to the `document` method. This allows users to call add with the same document ID as a document previously created, with the result that this will update the document's data in FireStore.

Reply:
I do see the merits of that. For the time being, I believe it's best for the plugin's API to match the underlying Android/iOS APIs as closely as possible. In this case, matching Android's [CollectionReference.add](https://firebase.google.com/docs/reference/android/com/google/firebase/firestore/CollectionReference.html#add(java.util.Map%3Cjava.lang.String,%20java.lang.Object%3E)), ObjC's FIRCollectionReference [addDocumentWithData](https://firebase.google.com/docs/reference/ios/firebasefirestore/api/reference/Classes/FIRCollectionReference#/c:objc(cs)FIRCollectionReference(im)addDocumentWithData:), and Swift's [CollectionReference.addDocument](https://firebase.google.com/docs/reference/swift/firebasefirestore/api/reference/Classes/CollectionReference#/c:objc(cs)FIRCollectionReference(im)addDocumentWithData:)